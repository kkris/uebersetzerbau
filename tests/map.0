// how to distinguish base case?
map = fun func -> fun array -> (func (head array)) . ((map func) (tail array)) end end ;

times2 = fun x -> 2 * x end;
main = fun array -> (map times2) array end;
