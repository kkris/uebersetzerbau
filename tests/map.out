ident map
=
fun
ident func
->
fun
ident array
->
fun
ident n
->
let
ident next
=
ident func
(
head
ident array
)
in
if
ident n
=
num 1
then
ident next
else
ident next
.
(
(
ident map
ident func
)
(
tail
ident array
)
)
(
ident n
-
num 1
)
end
end
end
end
end
;
ident times2
=
fun
ident x
->
num 2
*
ident x
end
;
ident main
=
fun
ident array
->
fun
ident n
->
(
(
ident map
ident times2
)
ident array
)
ident n
end
end
;
